Source/Dafny/Resolver.cs

```diff
@@ -3488,11 +3503,19 @@ public class ModuleBindings {
           } else if (member is Function) {
             var f = (Function)member;
             ResolveParameterDefaultValues_Pass1(f.Formals, f);
-            if (!f.IsGhost && f.Body != null) {
-              CheckIsCompilable(f.Body);
-            }
-            if (f.Body != null) {
-              DetermineTailRecursion(f);
+            if (f.ByMethodBody == null) {
+              if (!f.IsGhost && f.Body != null) {
+                CheckIsCompilable(f.Body, f);
+              }
+              if (f.Body != null) {
+                DetermineTailRecursion(f);
+              }
+            } else {
+              var m = f.ByMethodDecl;
+              Contract.Assert(m != null && !m.IsGhost);
+              ComputeGhostInterest(m.Body, m.IsGhost, m);
```

Comment:
Consider writing `false` instead of `m.IsGhost`, given the previous assert.

Reply:
done